src/vss_tools/vspec/vssexporters/vss2samm/helpers/vss_helper.py

```diff
@@ -155,85 +156,24 @@ def should_use_parent_prefix(vss_node: VSSNode) -> bool:
-            #               vss_node.data.instances = []
-            vss_node_instance_keywords = [
-                "Low",
-                "High",
-                "Left",
-                "Middle",
-                "Right",
-                "Front",
-                "Center",
-                "Rear",
-                "DriverSide",
-                "PassengerSide",
-                "Driver",
-                "Passenger",
-                "Primary",
-                "Secondary",
-                "FrontLeft",
-                "FrontMiddle",
-                "FrontRight",
-                "RearLeft",
-                "RearMiddle",
-                "RearRight",
-                "AnyPosition",
-            ]
-
-            # NOTE: Since there could be instance entries like: Row1, Row2... Row#
-            #       or Sensor1, Sensor2... Sensor#
-            #       which are more "dynamic", we use a RegExp to catch such entries.
-            #       Therefore they are not added in above collection.
-            rows_sensor_number_regexp = "((Row|Sensor){1}[0-9]+)"
-
-            # Expanded nodes have paths in the form: Vehicle.PARENT_NODE.Row1.DriverSide.NODE_NAME
-            # Normalize - shrink expanded vss_path to its normal form i.e. Vehicle.PARENT_NODE.NODE_NAME
-            # before to compare it with selected_signal_path
-            expanded_vss_path = vss_node_path
-            for step in expanded_vss_path.split("."):
-                if step in vss_node_instance_keywords or re.match(rows_sensor_number_regexp, step) is not None:
-                    # Remove only instance keywords
-                    vss_node_path = vss_node_path.replace(step, "")
-
-            # When expanded_steps is removed from vss_node_path,
-            # there might be remaining dots (.) at either start / end of the string
-            # or adjacent multiple dots within the path.
-            #
-            # For example: Vehicle.Cabin.Door...IsLocked or .Vehicle.Cabin.Door..Window.
-            #
-            # Make sure to clean these extra - dots (.) and put the path in normal format: Vehicle.Cabin.Door.IsLocked
-            vss_node_path = re.sub(r"\.{2,}", ".", vss_node_path)
-            if vss_node_path.startswith("."):
-                vss_node_path = vss_node_path[1:]
-            if vss_node_path.endswith("."):
-                vss_node_path = vss_node_path[:-1]
+    # NOTE: if the expand_opt is used in vss2samm exporter,
+    #       there will be required additional logic to make sure that "expanded" paths like:
+    #
+    #           Vehicle.Cabin.Door.Row1.DriverSide.Window.Position
+    #
+    #       would map to selected path:
+    #
+    #           Vehicle.Cabin.Door.Window.Position
 
+    for selected_signal_path in selected_signals_paths:
         # Check if current node is selected as usual
         if len(vss_node_path) > len(selected_signal_path):
```

Comment:
The idea was to check if current vss_path is part of the sdelected_path and vice versa, instead of if one starts with the other, but you are right.

I reworked this check as advised i.e. to check if one path starts with the other or vice versa.

Done.